business/istio_config.go

```diff
@@ -606,33 +611,79 @@ func (in *IstioConfigService) GetIstioConfigPermissions(namespaces []string) mod
 	istioConfigPermissions := make(models.IstioConfigPermissions, len(namespaces))
 
 	if len(namespaces) > 0 {
+		networkingPermissions := make(models.IstioConfigPermissions, len(namespaces))
+		securityPermissions := make(models.IstioConfigPermissions, len(namespaces))
+
 		wg := sync.WaitGroup{}
-		wg.Add(len(namespaces) * len(newIstioConfigTypes))
+		// We will query 2 times per namespace (networking.istio.io and security.istio.io)
+		wg.Add(len(namespaces) * 2)
 		for _, ns := range namespaces {
-			resourcePermissions := make(models.ResourcesPermissions, len(newIstioConfigTypes))
-			for _, rs := range newIstioConfigTypes {
-				resourcePermissions[rs] = &models.ResourcePermissions{
-					Create: false,
-					Update: false,
-					Delete: false,
+			networkingRP := make(models.ResourcesPermissions, len(newNetworkingConfigTypes))
+			securityRP := make(models.ResourcesPermissions, len(newSecurityConfigTypes))
+			networkingPermissions[ns] = &networkingRP
+			securityPermissions[ns] = &securityRP
+			/*
+				We can optimize this logic.
+				Instead of query all editable objects of networking.istio.io and security.istio.io we can query
+				only one per API, that will save several queries to the backend.
+
+				Synced with:
+				https://github.com/kiali/kiali-operator/blob/master/roles/default/kiali-deploy/templates/kubernetes/role.yaml#L62
```

Comment:
just FYI - this link will eventually grow stale because its linking to "master" rather than a fixed tag. Probably should have used "v1.36.0".